.github/workflows/build-docker.yaml

```diff
@@ -1,13 +1,57 @@
 name: Build docker
-on: push
```

Comment:
we shoudl put this as a reusable workflow. Perhaps good one to make an issue in .github repo

Reply:
It already is! I just didn't want to make too many changes at once. It's part of the nondefault_branch one in the templates.

Reply:
could we not use that then?

Reply:
Sure!

Reply:
[`2133210` (#192)](https://github.com/openclimatefix/uk-pvnet-app/pull/192/commits/213321068509fdb2f5e6f1f9297320de6cad87e6)